bin/git-forgit

```diff
@@ -1176,6 +1189,141 @@ _forgit_paths_list() {
+
+    opts="
+        $FORGIT_FZF_DEFAULT_OPTS
+        +s +m --tiebreak=index
+        --preview=\"$FORGIT worktree_preview {1}\"
+        $FORGIT_WORKTREE_SELECT_FZF_OPTS
+    "
+
+    tree=$(echo "$worktree_list" | FZF_DEFAULT_OPTS="$opts" fzf)
+    [[ -z "$tree" ]] && return 1
+    echo "$tree"
+}
+
+_forgit_worktree_lock_max_args() {
+    local usage
+    usage=$(git worktree lock --help 2>/dev/null | grep -E 'usage: git worktree lock' | head -n1)
+    usage=${usage#*:}
+     rest=$(echo "$usage" | awk '{for(i=4;i<=NF;++i)printf "%s ",$i; print ""}')
+    # shellcheck disable=SC2206
+    local args=($rest)
+    echo "${#args[@]}"
+}
+
+_forgit_git_worktree_lock() {
+    _forgit_worktree_lock_git_opts=()
+    _forgit_parse_array _forgit_worktree_lock_git_opts "$FORGIT_WORKTREE_LOCK_GIT_OPTS"
+    git worktree lock "${_forgit_worktree_lock_git_opts[@]}" "$@"
+}
+
+_forgit_worktree_lock() {
+    _forgit_inside_work_tree || _forgit_inside_git_dir || return 1
+
+    local tree opts max_args
+    max_args=$(_forgit_worktree_lock_max_args)
+
+    if [[ $# -ge "$max_args" ]] || { [[ $# -ne 0 ]] && _forgit_all_non_flags "$@"; }; then
+        git worktree lock "$@"
+        return $?
+    fi
+
+    opts="
+        $FORGIT_FZF_DEFAULT_OPTS
+        +s +m --tiebreak=index
+        --preview=\"$FORGIT worktree_preview {1}\"
+        $FORGIT_WORKTREE_LOCK_FZF_OPTS
+    "
+
+    tree=$(git worktree list | grep -vE "\(bare\)|locked" | awk '{print $1}' | FZF_DEFAULT_OPTS="$opts" fzf)
+    [[ -z "$tree" ]] && return 1
+    _forgit_git_worktree_lock "$tree" "$@"
+}
+
+_forgit_git_worktree_remove() {
+    _forgit_worktree_remove_git_opts=()
+    _forgit_parse_array _forgit_worktree_remove_git_opts "$FORGIT_WORKTREE_REMOVE_GIT_OPTS"
+    git worktree remove "${_forgit_worktree_remove_git_opts[@]}" "$@"
+}
+
+_forgit_worktree_remove() {
+    _forgit_inside_work_tree || _forgit_inside_git_dir || return 1
+    local worktree_list tree opts
+
+    if [[ $# -ne 0 ]] && _forgit_contains_non_flags "$@"; then
+        git worktree remove "$@"
+        return $?
+    fi
+
+    worktree_list=$(git worktree list | grep -v "(bare)")
+
+    count=$(echo "$worktree_list" | wc -l)
+    [[ $count -eq 1 ]] && return 1
+
+    opts="
+        $FORGIT_FZF_DEFAULT_OPTS
+        +s +m --tiebreak=index
+        --preview=\"$FORGIT worktree_preview {1}\"
+        $FORGIT_WORKTREE_REMOVE_FZF_OPTS
+    "
+
+    tree=$(echo "$worktree_list" | awk '{print $1}' | FZF_DEFAULT_OPTS="$opts" fzf)
+    [[ -z "$tree" ]] && return 1
+    _forgit_git_worktree_remove "$tree" "$@"
+}
+
+_forgit_worktree_unlock() {
+    _forgit_inside_work_tree || _forgit_inside_git_dir || return 1
+    if [[ $# -ne 0 ]]; then
+        git worktree unlock "$@"
+        worktree_unlock_status=$?
+        return $worktree_unlock_status
+    fi
+    local worktree_list tree opts
+
+    worktree_list=$(git worktree list | grep -v "(bare)" | grep -E "locked$")
+    count=$(echo "$worktree_list" | wc -l)
```

Comment:
`$count` is 1 when there are actually no locked workspaces because echo adds a new line. We can work around it like this:

```suggestion
    count=$(echo "$worktree_list" | grep -vcE '^$')
```

This applies to multiple functions.